### PR TITLE
pgScancodeWrapper now raises exception when trying to use __iter__

### PR DESCRIPTION
**Pull request by PurityLake**
#1696 

Added a tp_iter and tp_iternext field to pgScancodeWrapper that raise an exception when called.

This has actually already been merged as of earlier 11 hours ago so I'm unsure if this is already contained in this repo

### CODE REVIEW

**Review by MyreMylar — APPROVED** [review]
It is not in the repo yet. 

It looks good to me over here though too 👍

File: src_c/key.c [hunk 1]
```diff
@@ -88,6 +88,50 @@ pg_scancodewrapper_subscript(pgScancodeWrapper *self, PyObject *item)
...
+pg_iter_raise(PyObject *self)
+{
+    PyErr_SetString(PyExc_TypeError,
+                    "Iterating over key states is not supported");
+    return NULL;
+}
+
+/**
+ * There is an issue in PyPy that causes __iter__ to be called
+ * on creation of a ScandcodeWrapper. This stops this from
```
Comment (oddbookworm) [inline src_c/key.c:101]: slight typo here, inconsequential typo "ScandcodeWrapper"

Comment (PurityLake) [reply src_c/key.c]: I have fixed that type, thanks for pointint it out

File: docs/reST/ref/key.rst [hunk 2]
```diff
@@ -266,6 +266,11 @@ for ``KMOD_NONE``, which should be compared using equals ``==``). For example:
       translate these pushed keys into a fully translated character value. See
       the ``pygame.KEYDOWN`` events on the :mod:`pygame.event` queue for this
       functionality.
+   
+   .. versionadded:: 2.1.4
+      The collection of bools returned by ``get_pressed`` can not be iterated
+      over because the indexes of the internal tuple does not correpsond to the
```
Comment (oddbookworm) [inline docs/reST/ref/key.rst:272]: Nitpick:
"correspond" is misspelled here

File: docs/reST/ref/key.rst [hunk 3]
```diff
@@ -266,6 +266,11 @@ for ``KMOD_NONE``, which should be compared using equals ``==``). For example:
       translate these pushed keys into a fully translated character value. See
       the ``pygame.KEYDOWN`` events on the :mod:`pygame.event` queue for this
       functionality.
+   
+   .. versionadded:: 2.1.4
```
Comment (oddbookworm) [inline docs/reST/ref/key.rst:270]: Should this be a `versionchanged` instead of a `versionadded`?

**Review by oddbookworm — APPROVED** [review]
Couple small things that I'm not sure are worth requesting changes over. Other than those things, looks good to me!

**Comment by PurityLake** [comment]
@oddbookworm I want it to be a quality addition so if you have any issues feel free to say it

**Comment by oddbookworm** [comment]
> @oddbookworm I want it to be a quality addition so if you have any issues feel free to say it

If you want to go ahead and address my two comments above, I would appreciate that. Once that's done, I have no reservations about merging it

**Comment by PurityLake** [comment]
Those two issues are sorted now.

**Pull request merged**